ext/hash/hash.stub.php

```diff
@@ -60,6 +66,9 @@ function hash_hmac_algos(): array {}
  */
 function hash_pbkdf2(string $algo, #[\SensitiveParameter] string $password, string $salt, int $iterations, int $length = 0, bool $binary = false): string {}
 
+/**
+ * @compile-time-eval
```

Comment:
I don't consider it a good idea to evaluate `hash_equals()` at compile time. Its contract is that the execution time is a function of the input length and that would be violated for constant inputs.

It is likely rare already that the function is called with constant arguments, making the performance benefit very small. But if someone is doing it, they might have a reason for that (e.g. to simulate the expected delay in an error path) and thus evaluating it at compile time will violate their assumptions and possibly introduce a timing vulnerability.

Reply:
reverted